### PR TITLE
Convert setup-rust-container to composite action and simplify CI triggers

### PR DESCRIPTION
**Pull request by willbradshaw**
## Summary
- Convert `setup-rust-container` from a reusable workflow to a composite action, eliminating confusing duplicate "setup-rust / setup" jobs in the GitHub Actions UI
- Remove `workflow_run` triggers from integration tests (benchmark-* and test-chained), which were confusing and expensive

## Changes
1. **New composite action** at `.github/actions/setup-rust-container/action.yml`
2. **Updated 5 nf-test workflows** to use the action instead of the reusable workflow
3. **Removed `load_rust_container`** input from setup-nf-test action
4. **Deleted** `.github/workflows/setup-rust-container.yml` reusable workflow
5. **Simplified** benchmark and test-chained workflows to only use `pull_request` trigger

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by evanfields — APPROVED** [review]
Some changes requested, but I don't need to re-review. Let me know if you'll handle #641 on this branch.

File: .github/workflows/nf-test-modules.yml [hunk 1]
```diff
@@ -50,10 +48,15 @@ jobs:
         with:
           aws_access_key_id: ${{ secrets.NAO_TESTING_ACCESS_KEY }}
           aws_secret_access_key: ${{ secrets.NAO_TESTING_SECRET_ACCESS_KEY }}
-          load_rust_container: ${{ needs.setup-rust.outputs.rust-changed }}
+      - name: Setup Rust container
+        if: steps.filter.outputs.modules == 'true'
+        id: rust
+        uses: ./.github/actions/setup-rust-container
```
[the remark this answers is not in the document]
Comment (willbradshaw) [reply .github/workflows/nf-test-modules.yml]: Yeah, I think this is fine for the reason you state

File: .github/workflows/rust-tools.yml [hunk 2]
```diff
@@ -48,49 +46,29 @@ jobs:
...
-        uses: aws-actions/configure-aws-credentials@v4
-        with:
-          aws-access-key-id: ${{ secrets.NAO_TESTING_ACCESS_KEY }}
-          aws-secret-access-key: ${{ secrets.NAO_TESTING_SECRET_ACCESS_KEY }}
-          aws-region: us-east-1
-
-      # Scan the container for vulnerabilities before pushing (mirrors trivy-scan.yml for other containers)
+      # --- ECR push (only on push to dev/main, not PRs) ---
       - name: Install Trivy
+        if: steps.filter.outputs.rust == 'true' && github.event_name == 'push'
```
[the remark this answers is not in the document]
Comment (willbradshaw) [reply .github/workflows/rust-tools.yml]: We can't, because if the job is skipped entirely then the test won't pass. But we can separate out the ECR upload and vulnerability scanning into separate non-required jobs.

File: .github/workflows/benchmark-illumina-100M.yml [hunk 3]
```diff
@@ -1,36 +1,25 @@
...
-# This is a "release test" - see docs/ci.md for explanation of trigger strategy.
-# Uses workflow_run to sequence after rust-tools.yml, ensuring ECR has the latest container.
+# Release test - only runs on PRs to main/stable
 on:
   pull_request:
     branches: [main, stable, ci-test]
-  workflow_run:
-    workflows: ["Rust Tools CI"]
-    types: [completed]
-    branches: [dev]
```
[the remark this answers is not in the document]
Comment (willbradshaw) [reply .github/workflows/benchmark-illumina-100M.yml]: **Note: This comment is from Claude Code, not Will.**

I think there's a simple fix that eliminates this edge case entirely.

**Current logic:**
1. If Rust code matches `origin/main` → use ECR `:main`
2. Else if Rust code matches `origin/dev` → use ECR `:dev`
3. Else → build locally (cached by content hash)

**Proposed change:** Remove the `origin/dev` check entirely:
1. If Rust code matches `origin/main` → use ECR `:main`
2. Else → build locally (cached by content hash)

**Why this works:** The GitHub Actions cache is keyed by content hash, so the same Rust source always hits the same cache regardless of which PR/branch triggered the build. When you:
1. Open PR to dev with Rust changes → builds locally, cached
2. Merge to dev → rust-tools.yml pushes to ECR (but we don't rely on this)
3. Open PR dev→main → builds locally, **cache hit** from step 1

This eliminates any reliance on the `:dev` ECR tag being up-to-date, avoiding the race condition entirely. The tradeoff is slightly more local builds for PRs to dev, but they'll be cache hits after the first build anyway.

[the remark this answers is not in the document]
Comment (evanfields) [reply .github/workflows/benchmark-illumina-100M.yml]: I think Claude's comment is a bit confused vis a vis the release tests, which run on Batch and thus require a Wave/Fusion-able container, which means a container on ECR and not a local build. Unless I'm well misunderstanding, we never we running the release tests with local builds.

That said this hypothetical race condition is a super edge case, unlikely to occur in practice; I think ignoring or a 1-line addition to the docs are both fine and sufficient.

File: docs/ci.md
[the remark this answers is not in the document]
Comment (willbradshaw) [reply docs/ci.md]: True, will fix
